tests/test_multiple_databases.py

```diff
@@ -60,16 +61,15 @@ def test_add_objects(self):
         session.add(Bar('second bar'))
         session.commit()
 
-        foo_count = (
-            db.get_engine('foo')
-            .execute("SELECT COUNT(*) FROM foo")
-            .fetchone()[0]
-        )
-        bar_count = (
-            db.get_engine('bar')
-            .execute("SELECT COUNT(*) FROM bar")
-            .fetchone()[0]
-        )
+        with db.get_engine('foo').begin() as conn:
+            foo_count = conn.execute(
+                text("SELECT COUNT(*) FROM foo")
+            ).fetchone()[0]
+
+        with db.get_engine('bar').begin() as conn:
+            bar_count = conn.execute(
+                text("SELECT COUNT(*) FROM bar")
+            ).fetchone()[0]
```

Comment:
Just for me to understand this correctly, if we don't introduce this change in the test, then the test would fail in case SQLAlchemy 1.x is installed? Or is this how the call is made in the latest 1.x release?

Reply:
This is how the call should be written in 2.x.

Reply:
Hmm, in that case we'll need to write some compatibility code inside the main library. Otherwise we'll be breaking usage for 1.x users.

Reply:
That's a good point. I'm checking the https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#first-prerequisite-step-two-a-working-1-4-application to identify changes. Let me test it with the last 1.4 version.

Reply:
The tests passed with 1.4.9.
![Uploading Screenshot 2023-10-09 at 9.46.15 AM.png…]()

Reply:
Awesome, thanks for looking it up. In that case we should probably specify something like `SQLAlchemy >= 1.4` or something like that.